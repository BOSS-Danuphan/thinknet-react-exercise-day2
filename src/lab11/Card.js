import React from 'react'
const Card = (props) => (
  <div id="card" onClick={() => props.onClick({args1: '10', args2: '20'})}>
    <h1 id="title">{props.title}</h1>
    <p id="body">{props.body}</p>
  </div>
)
export default Card
