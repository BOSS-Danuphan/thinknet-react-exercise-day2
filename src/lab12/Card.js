import React from 'react'
class Card extends React.Component {
  state = {
    page: 1,
    pageSize: 20,
    rendered: false
  }

  componentDidMount() {
    this.setState({rendered: true})
    this.props.callApi();
  }

  render() {
    const props = this.props
    return (
      <div id="card">
        <h1 id="title">{props.title}</h1>
        <p id="body">{props.body}</p>
        <p id="page">{this.state.page}</p>
      </div>
    )
  }
}

export default Card
