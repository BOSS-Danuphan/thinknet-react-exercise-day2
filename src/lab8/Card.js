import React from 'react'
const Card = () => (
  <div className="find-me">
    <h1 id="title">My Title</ h1>
    <p id="body">body</p>
  </div>
)
export default Card