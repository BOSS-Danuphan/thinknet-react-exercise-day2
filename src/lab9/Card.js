import React from 'react'
const Card = (props) => (
  <div className="find-me">
    <h1 id="title">{props.title || 'Default Title'}</ h1>
    <p id="body">body</p>
  </div>
)
export default Card