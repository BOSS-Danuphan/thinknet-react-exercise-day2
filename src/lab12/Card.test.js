import React from 'react'
import { shallow } from 'enzyme'
import sinon from 'sinon'
import Card from './Card'

const props = {callApi: sinon.spy()}

describe('Test initial state', () => {
  test('Default state rendered should be false as default', () => {
    const wrapper = shallow(<Card {...props} />, { disableLifecycleMethods: true });
    expect(wrapper.state()).toHaveProperty('rendered', false);
    wrapper.setState({page: 10})
    
    expect(props.callApi.notCalled).toBeTruthy()
  });

  test('Default state rendered should be false as default', () => {
    const wrapper = shallow(<Card {...props} />);
    expect(wrapper.state()).toHaveProperty('rendered', true);
    expect(props.callApi.callCount).toBe(1)
  });

  test('Set state when disable life cycle', () => {
    const wrapper = shallow(<Card {...props} />, { disableLifecycleMethods: true });
    expect(wrapper.state()).toHaveProperty('rendered', false);
    wrapper.setState({page: 10})
    expect(wrapper).toMatchSnapshot()
  });
});