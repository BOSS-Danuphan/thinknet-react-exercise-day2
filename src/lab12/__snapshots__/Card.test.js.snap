// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test initial state Set state when disable life cycle 1`] = `
<div
  id="card"
>
  <h1
    id="title"
  />
  <p
    id="body"
  />
  <p
    id="page"
  >
    10
  </p>
</div>
`;
